sealtk/core/test/AutoLevelsTask.cpp

```diff
@@ -0,0 +1,113 @@
+/* This file is part of SEAL-TK, and is distributed under the OSI-approved BSD
+ * 3-Clause License. See top-level LICENSE file or
+ * https://github.com/Kitware/seal-tk/blob/master/LICENSE for details. */
+
+#include <sealtk/test/TestCore.hpp>
+
+#include <sealtk/core/AutoLevelsTask.hpp>
+
+#include <vital/algo/image_io.h>
+#include <vital/plugin_loader/plugin_manager.h>
+
+#include <qtStlUtil.h>
+
+#include <QObject>
+#include <QtTest>
+
+namespace kv = kwiver::vital;
+
+namespace sealtk
+{
+
+namespace core
+{
+
+namespace test
+{
+
+// ============================================================================
+class TestAutoLevelsTask : public QObject
+{
+  Q_OBJECT
+
+private slots:
+  void initTestCase();
+  void levels();
+  void levels_data();
+
+private:
+  kv::image_container_sptr m_image;
+};
+
+// ----------------------------------------------------------------------------
+void TestAutoLevelsTask::initTestCase()
+{
+  kv::plugin_manager::instance().load_all_plugins();
+  if (auto io = kv::algo::image_io::create("vxl"))
+  {
+    m_image = io->load(
+      stdString(SEALTK_TEST_DATA_PATH("AutoLevelsTask/test.tif")));
+  }
+}
+
+// ----------------------------------------------------------------------------
+void TestAutoLevelsTask::levels()
+{
+  // Check that test image was loaded (pointless to proceed otherwise!)
+  QVERIFY(m_image);
+
+  // Fetch test case parameters
+  QFETCH(double, outlierDeviance);
+  QFETCH(double, outlierTolerance);
+  QFETCH(float, expectedLow);
+  QFETCH(float, expectedHigh);
+
+  // Set up the task
+  auto computedLow = 0.0f;
+  auto computedHigh = 1.0f;
+  AutoLevelsTask task{m_image, outlierDeviance, outlierTolerance};
+
+  connect(&task, &AutoLevelsTask::levelsUpdated,
+          [&](float l, float h){
+            qDebug() << "result:" << l << h;
+            computedLow = l;
+            computedHigh = h;
+          });
```

Comment:
Could you instead emit a vector with one high/low pair for each iteration? That way we can make sure every stage is correct instead of just the last one.